app/core/tests/test_api.py

```diff
@@ -722,3 +725,90 @@ def test_project_stack_element_workflow(auth_client):
     assert len(res_list.data) == 1
     assert res_list.data[0]["project"] == project_uuid
     assert res_list.data[0]["stack_element"] == stack_element_uuid
+
+
+def test_create_url_status_type(auth_client):
+    payload = {"name": "active", "description": "URL is live"}
+    res = auth_client.post(URL_STATUS_TYPES_URL, payload)
+    assert res.status_code == status.HTTP_201_CREATED
+    assert res.data["name"] == "active"
+    assert UrlStatusType.objects.filter(uuid=res.data["uuid"]).exists()
+
+
+def test_list_url_status_types(auth_client):
+    # ensure at least one exists
+    auth_client.post(URL_STATUS_TYPES_URL, {"name": "archived"})
+    res = auth_client.get(URL_STATUS_TYPES_URL)
+    assert res.status_code == status.HTTP_200_OK
+    assert any(item["name"] in ("active", "archived") for item in res.data)
+
+
+def test_project_url_accepts_status_type(
+    auth_client, project, url_type, url_status_type
+):
+    payload = {
+        "project": project.pk,
+        "url_type": url_type.pk,
+        "name": "Readme",
+        "external_id": "",
+        "url": "https://example.com/readme",
+        "url_status_type": str(url_status_type.uuid),
```

Comment:
Are you seeing errors if you don't convert it to string? I think it's already string in most cases.

Also, we've been using `.pk` to get the primary key, which works whether the model is using an `id` field or a `uuid` field as primary key.

One of these days, we'll create an issue and PR to change all the `.uuid` code over to `.pk`.